monkey/monkey_island/setup/island_config_options.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from monkey_island.cc.server_utils.consts import (
+    DEFAULT_DATA_DIR,
+    DEFAULT_LOG_LEVEL,
+    DEFAULT_START_MONGO_DB,
+)
+
+
+@dataclass
+class IslandConfigOptions:
+    log_level = DEFAULT_LOG_LEVEL
+    data_dir = DEFAULT_DATA_DIR
+    start_mongodb = DEFAULT_START_MONGO_DB
+
+    @staticmethod
+    def build_from_config_file_contents(config_contents: dict) -> IslandConfigOptions:
+        config = IslandConfigOptions()
+        if "data_dir" in config_contents:
+            config.data_dir = config_contents["data_dir"]
+
+        if "log_level" in config_contents:
+            config.log_level = config_contents["log_level"]
```

Comment:
For these `if` statements, you can use `get()` instead of setting the default values on the class variables.

```python
config.data_dir = config_contents.get("data_dir", DEFAULT_DATA_DIR)
```

See [python antipatterns](https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html#not-using-get-to-return-a-default-value-from-a-dict) for more information.

Reply:
nice

Reply:
I've tried this option, but IMO it doesn't make the code more readable (doesn't fit the antipattern exactly). The cons are:
1. We can't move defaults from the dataclass definition into this method, because island config options are used without this constructor. So we end up defining the same defaults at two places.
2. I had to do 
```
config.start_mongodb = config_contents.\
            get(["mongodb"], {"start_mongodb": DEFAULT_START_MONGO_DB}).\
            get("start_mongodb", DEFAULT_START_MONGO_DB)
```
IMO this is not more readable than what we currently have.

Reply:
Yeah, I wasn't suggesting it for the mongodb case, just the other two. But if you want to leave them so all three cases are consistent, I'm ok with it.

I'm not clear on why we need the default constructor provided by `@dataclass`, but we'll talk about it on slack.